.github/workflows/push-image.yaml

```diff
@@ -0,0 +1,82 @@
+---
+name: Build and push image to ECR
+
+on:
+  workflow_dispatch:
+    inputs:
+      git_sha:
+        description: "Enter the git sha which you would like to build"
+        required: true
+        type: string
+        default: 42d2005d04b59d14aacf4eef5ccee353e9a531b7
+      docker_base_distribution:
+        description: "Enter the Docker Base Distribution (bionic, focal, stretch, buster, or bullseye)"
+        required: true
+        type: string
+        default: bullseye
+      network_name:
+        description: "Enter the Network Name (berkeley or mainnet)"
+        required: true
+        type: string
+        default: mainnet
+      docker_base_tag_prefix:
+        description: "Enter the Base Tag Prefix: See https://github.com/MinaProtocol/mina/discussions/13056 for more details"
+        required: true
+        type: string
+env:
+  IMAGE_BASE_COMMIT: ${{ github.event.inputs.git_sha }}
+  DOCKER_BASE_DISTRIBUTION: ${{ github.event.inputs.docker_base_distribution }}
+  NETWORK_NAME: ${{ github.event.inputs.network_name }}
+  DOCKER_BASE_TAG_PREFIX: ${{ github.event.inputs.docker_base_tag_prefix }}
+  FUNCTIONS: ./docker-image-ecr/functions.sh
+  VARIABLES: ./docker-image-ecr/variables.sh
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  build-docker-image:
+
+    name: Build and push Docker Image
+    runs-on: minafoundation-default-runners
+
+    steps:
+
+      - name: 📥 Checkout
+        uses: actions/checkout@v3    
+
+      - name: 📥 Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: ⚙ Shorten Commit SHA
+        run: |
+          source ${{ env.FUNCTIONS }}
+          echo "IMAGE_BASE_COMMIT_SHORT=$(shorten_commit_sha $IMAGE_BASE_COMMIT)" >> $GITHUB_ENV
+
+      - name: ⚙ Check ECR if Tag already exists
+        run: |
+          source ${{ env.FUNCTIONS }}
+          TAG=${{ env.DOCKER_BASE_TAG_PREFIX }}-${{ env.IMAGE_BASE_COMMIT_SHORT }}-${{ env.DOCKER_BASE_DISTRIBUTION }}-${{ env.NETWORK_NAME }}
```

Comment:
Seems that `$TAG` can be also added to `GITHUB_ENV` so it can be reused on subsequent steps.